spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -134,14 +134,8 @@ class CodeEditor(TextEditBaseWidget):
     UPDATE_DECORATIONS_TIMEOUT = 500  # milliseconds
 
     # Timeouts (in milliseconds) to sychronize symbols and folding after
-    # linting results arrive, according to the number of lines in the file.
-    SYNC_SYMBOLS_AND_FOLDING_TIMEOUTS = {
-        # Lines: Timeout
-        500: 350,
-        1500: 800,
-        2500: 1200,
-        6500: 1800
-    }
+    # linting results arrive
+    SYNC_SYMBOLS_AND_FOLDING_TIMEOUT = 500  # milliseconds

```

Comment:
As this constant's name says, this controls updating folding **and** symbols. So removing the timeouts we have for file size will introduce sluggishness when people have the outline pane visible.

So please revert this change and the others related to it.

Reply:
I don't find any affect when the outline pane is visible. Updating the outline explorer pane seems to take less than 150ms when viewing the `codeeditor.py` (5000+ lines, 250+ definitions) and did not affect typing latency in the editor in my tests. I can revert, but I don't think it's necessary. If there is concern for older, slower systems, I think keeping a single timeout may still be better (simpler) and just increasing the timeout (1000ms?).

Reply:
This is not correct on my tests. In the gifs below I disabled folding and only have the outline visible so we can have a fair comparison between 5.x and this PR:

**This PR**

![slow](https://user-images.githubusercontent.com/365293/134980483-5fdee99a-ff0d-4351-b29d-040818644465.gif)

**5.x**

![fast](https://user-images.githubusercontent.com/365293/134980588-839aeec2-f6ef-4cb5-84f3-73d372390809.gif)